core/IO.carp

```diff
@@ -3,62 +3,64 @@
   (register print (Fn [(Ref String)] ()))
-  (doc errorln "prints a string ref to stderr, appends a newline.")
+  (doc      errorln "prints a string ref to stderr, appends a newline.")
   (register errorln (Fn [(Ref String)] ()))
-  (doc error "prints a string ref to stderr, does not append a newline.")
+  (doc      error "prints a string ref to stderr, does not append a newline.")
   (register error (Fn [(Ref String)] ()))
-  (doc get-line "gets a line from stdin.")
+  (doc      get-line "gets a line from stdin.")
   (register get-line (Fn [] String))
-  (doc get-char "gets a character from stdin.")
+  (doc      get-char "gets a character from stdin (thin wrapper for getchar() from C standard library).")
   (register get-char (Fn [] Char) "getchar")
-  (doc read-file "returns the contents of a file passed as argument as a string.")
-  (register read-file (Fn [&String] String))
-  (doc exit "exit the current program with a return code.")
+  (doc      exit "exit the current program with a return code (thin wrapper for the C standard library).")
   (register exit (Fn [Int] ()) "exit")
+  (doc      EOF "the End-Of-File character as a literal (thin wrapper for the C standard library)")
   (register EOF Char)
-  (doc EOF "the End-Of-File character as a literal.")
-  (doc fopen "opens a file by name using a mode (one or multiple of [r]ead, [w]rite, and [a]ppend), returns a file pointer. Consider using the function open-file instead.")
+  (doc      fopen "opens a file for input/output/appending (thin wrapper for fopen(pathname, mode) from C standard library). Consider using the function open-file instead.")
   (register fopen (Fn [&String &String] (Ptr FILE)))
-
-  (doc open-file "opens a file by name using a mode (one or multiple of [r]ead, [w]rite, and [a]ppend), returns a Result type that contains an error string or a file pointer.")
+  (doc open-file "opens a file by name using a mode (e.g. [r]ead, [w]rite, [a]ppend),  [rb] read binary...). See fopen() in the C standard library for a detailed description of valid parameters.")
   (defn open-file [filename mode]
     (let [ptr (IO.fopen filename mode)]
       (if (null? ptr)
         (do
           (Result.Error System.errno))
         (Result.Success ptr))))
-
-  (doc fclose "closes a file pointer.")
+  (doc      fclose "closes a file pointer (thin wrapper for the C standard library).")
   (register fclose (Fn [(Ptr FILE)] ()))
-  (doc fgetc "gets a character from a file pointer.")
+  (doc      fgetc "gets a character from a file pointer (thin wrapper for the C standard library).")
   (register fgetc (Fn [(Ptr FILE)] Char))
-  (doc fwrite "writes to a file pointer.")
-  (register fwrite (Fn [a Int Int (Ptr FILE)] ()) "fwrite")
-  (doc fread "reads from a file pointer into a pointer.")
+  (doc      fwrite "writes a C-string to a file and returns the number of item written items (thin wrapper for fwrite(cstr, item-size, items-count, file) from C standard library). Consider using write-file instead.")
```

Comment:
typo: s/`item written items`/`written items`

Reply:
Thank you, good catch. :-)
Done